django/test/signals.py

```diff
@@ -172,3 +173,24 @@ def auth_password_validators_changed(**kwargs):
 def user_model_swapped(**kwargs):
     if kwargs['setting'] == 'AUTH_USER_MODEL':
         apps.clear_cache()
+        try:
+            from django.contrib.auth import get_user_model
+            UserModel = get_user_model()
+        except ImproperlyConfigured:
+            # Some tests set an invalid AUTH_USER_MODEL.
+            pass
+        else:
+            from django.contrib.auth import backends
```

Comment:
Oops, I retract that. Third-party packages that cache the value of `get_user_model()`, typically in a module level variable, will have to invalidate that cache by listening the signal and writing code similar to this if they want to change `AUTH_USER_MODEL` in tests.